agentconnection.cpp

```diff
@@ -1,8 +1,93 @@
 #include "agentconnection.h"
 #include "message.h"
+#include "allocations.h"
 
+#include <QtEndian>
 #include <QTimer>
 
+Entity::Entity(int ronum, const char *data, int length) : RoutingObject(ronum, data, length)
+{
+    QByteArray sk;
+    if (ronum == bwpo::num::ROEntityWKey)
+    {
+        this->offset = 32;
+        sk = QByteArray::fromRawData(data, this->offset);
+        ronum = bwpo::num::ROEntity;
+    }
+
+    Q_ASSERT(ronum == bwpo::num::ROEntity);
+
+    const char* content = this->content();
+    this->vk = QByteArray::fromRawData(data, 32);
+
+    int idx = 32;
+    int ln;
+
+    forever
+    {
+        switch (content[idx])
+        {
+        case 0x02: // Creation date
+            Q_ASSERT_X(content[idx + 1] == 8, "Entity", "invalid creation date");
```

Comment:
Good point. Trouble this, this is a constructor so I can't return an error. I'll probably add a "valid" boolean and set it to false if I detect a problem (like out-of-bounds array access) while decoding.